packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts

```diff
@@ -0,0 +1,173 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { IContainer } from "@fluidframework/container-definitions";
+import { ISequencedDocumentMessage, ISummaryTree, SummaryType } from "@fluidframework/protocol-definitions";
+import { IContainerRuntime } from "@fluidframework/container-runtime-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import {
+    ITestObjectProvider,
+    createSummarizer,
+    summarizeNow,
+    waitForContainerConnection,
+} from "@fluidframework/test-utils";
+import { describeNoCompat, ITestDataObject, TestDataObjectType } from "@fluidframework/test-version-utils";
+import { defaultGCConfig } from "./gcTestConfigs";
+import {
+    getGCStateFromSummary,
+} from "./gcTestSummaryUtils";
+
+/**
+ * Validates that that reference state of nodes is correct irrespective of whether a summarizer loads from the
+ * latest summary or an older summary.
+ */
+describeNoCompat("GC loading from older summaries", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    let mainContainer: IContainer;
+    let containerRuntime: IContainerRuntime;
+    let dataStoreA: ITestDataObject;
+
+    /**
+     * Submits a summary and returns the reference state for all the nodes in the container.
+     * If a node is referenced, its value is true. If it's unreferenced, its value is false.
+     * @returns a map of nodePath to its unreferenced timestamp.
+     */
+    async function getReferenceState(summaryTree: ISummaryTree) {
+        const gcState = getGCStateFromSummary(summaryTree);
+        assert(gcState !== undefined, "GC tree is not available in the summary");
+
+        const nodeReferenceStates: Map<string, boolean> = new Map();
```

Comment:
Thought: `nodeIsReferencedMap`? and then the if statement below could be
`nodeIsReferencedMap.set(nodePath.slice(1), nodeData.unreferencedTimestampMs === undefined ? true: false);`